__tests__/app.test.js

```diff
@@ -132,22 +123,61 @@ describe("GET /api/reviews/:review_id/comments", () => {
 				expect(body.comments).toEqual([]);
 			});
 	});
-  
+
 	test("404 Not found if review_id isn't in the database", () => {
 		return request(app)
 			.get("/api/reviews/78/comments")
 			.expect(404)
-			.then(({ body }) => {
-				expect(body).toEqual({ msg: "Not found" });
-			});
+			.then(({ body }) => expect(body).toEqual({ msg: "Not found" }));
 	});
 	test("400 If key is not the correct data type", () => {
 		return request(app)
 			.get("/api/reviews/hello/comments")
 			.expect(400)
+			.then(({ body }) => expect(body).toEqual({ msg: "Bad Request" }));
+	});
+});
+
+describe("POST /api/reviews/:review_id/comments should post a new comment when given a body", () => {
+	test("201 should add a new comment to the comments database", () => {
+		return request(app)
+			.post("/api/reviews/2/comments")
+			.send({ author: "mallionaire", body: "Great for beginners" })
+			.expect(201)
 			.then(({ body }) => {
-				expect(body).toEqual({ msg: "Bad Request" });
+				expect(body.comment[0]).toHaveProperty("comment_id");
+				expect(body.comment[0]).toHaveProperty("body");
+				expect(body.comment[0]).toHaveProperty("review_id");
+				expect(body.comment[0]).toHaveProperty("author");
+				expect(body.comment[0]).toHaveProperty("created_at");
+				expect(body.comment[0].review_id).toEqual(2);
+				expect(body.comment[0].author).toEqual("mallionaire");
+				expect(body.comment[0].body).toEqual("Great for beginners");
 			});
 	});
+	test("404 should return an error if given a bad review_id", () => {
+		return request(app)
+			.post("/api/reviews/9001/comments")
+			.send({ author: "mallionaire", body: "Best wardrobe I've ever used" })
+			.expect(404)
+			.then(({ body }) => expect(body).toEqual({ msg: "Not found" }));
+	});
+	test("400 if datatype of the reviewId is wrong", () => {
+		return request(app)
+			.post("/api/reviews/hello/comments")
+			.send({ author: "mallionaire", body: "Hello Granddad!" })
+			.expect(400)
+			.then(({ body }) => expect(body).toEqual({ msg: "Bad Request" }));
+	});
+	test("400 If the body is in an unacceptable format", () => {
+		return request(app)
+			.post("/api/reviews/3/comments")
+			.send({
+				person: "Haz",
+				msg: "Hello, I am still waiting for my delivery of oranges",
+			})
+			.expect(400)
+			.then(({ body }) => expect(body).toEqual({ msg: "Bad Request" }));
+	});
```

Comment:
Testing looks great!
What would happen if the request body was sound, but the author didn't exist?